tools/dependency/release_dates.py

```diff
@@ -40,32 +42,58 @@ def verify_and_print_latest_release(dep, repo, metadata_version, release_date):
         latest_release = repo.get_latest_release()
         if latest_release.created_at > release_date and latest_release.tag_name != metadata_version:
```

Comment:
nit: i would move all of this logic after the `try...except` and return in the `except` clause

afaict none of this logic is related to the github exception and it means this chunk can be dedented